apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProvider.java

```diff
@@ -347,12 +348,17 @@ public Bundle findByPatient(
           paging, eobs, loadedFilterManager.getTransactionTime());
     }
 
+    // See if we have claims data for the beneficiary.
+    BitSet bitSet = QueryUtils.hasClaimsData(entityManager, beneficiaryId);
+    LOGGER.info(
+        String.format("BitSet for V2 claims, bene_id %d: %s", beneficiaryId, bitSet.toString()));
+
     /*
      * The way our JPA/SQL schema is setup, we have to run a separate search for
      * each claim type, then combine the results. It's not super efficient, but it's
      * also not so inefficient that it's worth fixing.
      */
-    if (claimTypes.contains(ClaimTypeV2.CARRIER)) {
+    if (claimTypes.contains(ClaimTypeV2.CARRIER) && bitSet.get(QueryUtils.CARRIER_HAS_DATA)) {
```

Comment:
These ifs could be a single loop if we used an `EnumSet`.

Reply:
yup. will change to EnumSet when we do the parallel work.